pkg/devconsole/proxy/proxy.go

```diff
@@ -48,9 +62,9 @@ func serve(r *http.Request, user *auth.User) (ProxyResponse, error) {
 
 	var serviceRequest *http.Request
 	if request.Body == "" {
-		serviceRequest, err = http.NewRequest(request.Method, request.Url, nil)
+		serviceRequest, err = http.NewRequestWithContext(r.Context(), request.Method, request.Url, nil)
 	} else {
-		serviceRequest, err = http.NewRequest(request.Method, request.Url, strings.NewReader(request.Body))
+		serviceRequest, err = http.NewRequestWithContext(r.Context(), request.Method, request.Url, strings.NewReader(request.Body))
```

Comment:
This allows the server request to inherit the client's request's context. In this way, if the client (UI) cancels its request the server request will also be cancelled, freeing the active connection and also cancelling the request at the target server (e.g. Tekton Results). This should cause a cascading performance improvement